lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -504,10 +531,10 @@ def write(chunk)
             labels_utf8_pairs = labels.map do |k, v|
               [k.encode('utf-8'), convert_to_utf8(v)]
             end
-            utf8_log_name = convert_to_utf8(log_name)
 
+            log_name = ERB::Util.url_encode(log_name)
```

Comment:
Yes, `log_name` could be extracted from `common_labels`. I've added sanitization there as well (near `def log_name`)